x/concentrated-liquidity/pool_hooks_test.go

```diff
@@ -0,0 +1,187 @@
+package concentrated_liquidity_test
+
+import (
+	"encoding/json"
+	"os"
+
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v20/x/concentrated-liquidity/types"
+)
+
+var (
+	validCosmwasmAddress   = "osmo14hj2tavq8fpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9sq2r9g9"
+	invalidCosmwasmAddress = "osmo1{}{}4hj2tfpesdwxxcu44rty3hh90vhujrvcmstl4zr3txmfvw9sq2r9g9"
+	validActionPrefix      = "beforeSwap"
+	counterContractPath    = "./testcontracts/compiled-wasm/counter.wasm"
+)
+
+// Message structs for the test CW contract, which is simply a counter
+// that counts `Amount` times and does a state write on each iteration.
+type CountMsg struct {
+	Amount int64 `json:"amount"`
+}
+type CountSudoMsg struct {
+	Count CountMsg `json:"count"`
+}
+
+// TestSetAndGetPoolHookContract tests the basic functionality of setting and getting a CW contract address for a specific hook type from state.
+func (s *KeeperTestSuite) TestSetAndGetPoolHookContract() {
```

Comment:
Please test deletion

Reply:
Added tests for these as well and related edge case (e.g. attempting to delete when no contract was set)